add `in-memory-engine` config items (#21044)
pingcap/docs: releases/release-8.5.0.md

| -------- | -------- | -------- | -------- |
| TiDB | [`deprecate-integer-display-length`](/tidb-configuration-file.md#deprecate-integer-display-length) | Modified | Starting from v8.5.0, the integer display width feature is deprecated. The default value of this configuration item is changed from `false` to `true`. |
| TiKV | [`raft-client-queue-size`](/tikv-configuration-file.md#raft-client-queue-size) | Modified | Changes the default value from `8192` to `16384`. |
| TiKV | [`in-memory-engine.capacity`](/tikv-configuration-file.md#capacity-new-in-v850) | Newly added | Controls the maximum memory size that the in-memory engine can use. The default value is 10% of the system memory. The maximum value is 5 GiB.|
| TiKV | [`in-memory-engine.enable`](/tikv-configuration-file.md#enable-new-in-v850) | Newly added | Controls whether to enable the in-memory engine to accelerate multi-version queries. The default value is `false`, which means that the in-memory engine is disabled. |

> | TiKV | [`in-memory-engine.gc-run-interval`](/tikv-configuration-file.md#gc-run-interval-new-in-v850) | Newly added | Controls the time interval for the in-memory engine to GC the cached MVCC versions. The default value is `"3m"`.|

Reviewer: ```suggestion
| TiKV | [`in-memory-engine.gc-run-interval`](/tikv-configuration-file.md#gc-run-interval-new-in-v850) | Newly added | Controls the time interval at which the in-memory engine performs garbage collection (GC) on cached MVCC versions. The default value is `"3m"`.|
```
